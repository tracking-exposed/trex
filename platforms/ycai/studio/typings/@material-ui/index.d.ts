// import  * as React from 'react';
import { Palette } from '@material-ui/core/styles/createPalette';
// import type { Theme } from '@material-ui/core/styles/createTheme';

// declare module '@material-ui/core/styles/createTheme' {
//   export interface Theme {
//     // status: {
//     //   danger: React.CSSProperties['color'];
//     // };
//   }
//   export interface ThemeOptions {
//     // status: {
//     //   danger: React.CSSProperties['color'];
//     // };
//   }
// }

declare module '@material-ui/core/styles/createPalette' {
  interface Palette {
    violet: Palette['primary'];
    yellow: Palette['primary'];
  }

  interface PaletteOptions {
    violet: PaletteOptions['primary'];
    yellow: PaletteOptions['primary'];
  }
}
