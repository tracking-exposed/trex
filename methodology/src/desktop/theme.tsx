import { createTheme } from "@material-ui/core";

export const theme = createTheme();
