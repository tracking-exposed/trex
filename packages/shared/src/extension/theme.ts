import { createTheme } from '@material-ui/core/styles';

// define @mui theme for popup
export const theme = createTheme({
  typography: {
    fontFamily: 'Trex-Regular',
  },
});
