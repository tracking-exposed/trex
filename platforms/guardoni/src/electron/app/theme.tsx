import { createTheme } from '@material-ui/core';
import './global.css';

const primaryColor = '#23AA9A';
const black = '#000000';

export const theme = createTheme({
  typography: {
    fontSize: 16,
    fontFamily: 'Trex',
    fontWeightRegular: 400,
    fontWeightMedium: 600,
    fontWeightBold: 800,
    h1: {
      fontWeight: 800,
      fontSize: '4rem',
      marginBottom: 32,
    },
    h2: {
      fontWeight: 800,
      fontSize: '3.6rem',
      marginBottom: 16,
    },
    h3: {
      fontWeight: 800,
      fontSize: '1.8rem',
      marginBottom: 10,
    },
    h4: {
      fontWeight: 600,
      fontSize: '1.6rem',
      marginBottom: 8,
    },
    h5: {
      fontWeight: 800,
      fontSize: '1.3rem',
    },
    h6: {
      fontWeight: 400,
      fontSize: '1.1rem',
      marginBottom: 4,
    },
    subtitle1: {
      fontSize: '1rem',
      fontWeight: 300,
      lineHeight: 1.4,
      marginBottom: 20,
    },
    subtitle2: {
      fontSize: '0.9rem',
      fontWeight: 800,
      lineHeight: 1.1,
    },
    body1: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: 1.1,
      color: black,
    },
    body2: {
      fontWeight: 400,
      fontSize: '0.8rem',
      lineHeight: 1.2,
    },
    caption: {
      fontSize: '0.7rem',
      lineHeight: 1.3,
    },
  },
  palette: {
    primary: {
      main: primaryColor,
    },
  },
});
