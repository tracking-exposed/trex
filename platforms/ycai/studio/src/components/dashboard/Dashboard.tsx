import React from 'react';
import {
  Grid,
  Link,
  Typography,
  useTheme,
  Divider,
  Box,
} from '@material-ui/core';
import { ArrowBack as ArrowBackIcon } from '@material-ui/icons';
import { makeStyles } from '@material-ui/core/styles';
import * as QR from 'avenger/lib/QueryResult';
import { declareQueries } from 'avenger/lib/react';
import { pipe } from 'fp-ts/lib/function';
import { useTranslation } from 'react-i18next';
import {
  accountLinkCompleted,
  auth,
  localProfile,
} from '../../state/dashboard/creator.queries';
import {
  CurrentView,
  currentView,
  doUpdateCurrentView,
} from '../../utils/location.utils';
import { ErrorBox } from '@shared/components/Error/ErrorBox';
import { LazyFullSizeLoader } from '../common/FullSizeLoader';
import Settings from './Settings';
import { AnalyticsPage } from './community/AnalyticsPage';
import { LinkAccount } from './LinkAccount';
import { Sidebar } from './Sidebar';
import { Lab } from './lab/Lab';
import { LabVideoEdit } from './lab/LabVideoEdit';
import { ContentCreator } from '@shared/models/ContentCreator';
import { AuthResponse } from '@shared/models/Auth';
import GemCollection from './lab/GemCollection';
import Congrats from './account/congrats';
import HelperPopover from '../common/HelperPopover';

const useStyles = makeStyles((theme) => ({
  root: {
    margin: 0,
    width: '100%',
    minHeight: '100%',
    backgroundColor: theme.palette.background.default,
  },
  labEditTitle: {
    whiteSpace: 'pre-line',
    paddingTop: theme.spacing(1),
    display: 'flex',
    alignItems: 'center',
    '& a': {
      color: theme.palette.common.black,
      marginTop: 3,
      marginRight: theme.spacing(1),
      '&:hover': {
        cursor: 'pointer',
      },
    },
  },
  congratsTitle: {
    whiteSpace: 'pre-line',
    paddingTop: theme.spacing(10),
    fontSize: theme.spacing(5),
    fontWeight: theme.typography.h1.fontWeight,
    lineHeight: 1.4,
    display: 'flex',
    alignItems: 'center',
    marginBottom: theme.spacing(3),
    '& a': {
      color: theme.palette.common.black,
      marginTop: 3,
      marginRight: theme.spacing(1),
      '&:hover': {
        cursor: 'pointer',
      },
    },
  },
}));

interface DashboardContentProps {
  currentView: CurrentView;
  profile: ContentCreator | null;
  auth: AuthResponse | null;
  accountLinkCompleted: boolean;
}

const DashboardContent: React.FC<DashboardContentProps> = ({
  currentView,
  profile,
  auth,
  accountLinkCompleted,
}) => {
  const { t } = useTranslation();
  const theme = useTheme();
  const classes = useStyles();

  const [
    currentViewLabel,
    currentViewSubtitle,
    helperText,
    currentViewContent,
  ] = React.useMemo(() => {
    switch (currentView.view) {
      case 'settings':
        // eslint-disable-next-line react/jsx-key
        return [t('routes:settings'), '', null, <Settings />];
      default: {
        if (profile === null) {
          return [
            t('routes:link_account'),
            t('link_account:subtitle'),
            null,
            // eslint-disable-next-line react/jsx-key
            <LinkAccount auth={auth} />,
          ];
        }

        if (!accountLinkCompleted) {
          return [
            t('routes:congrats'),
            t('congrats:subtitle'),
            null,
            <Congrats profile={profile} />,
          ];
        }

        switch (currentView.view) {
          case 'labEdit':
            return [
              t('routes:lab_edit_title'),
              t('routes:lab_edit_subtitle'),
              null,
              // eslint-disable-next-line react/jsx-key
              <LabVideoEdit videoId={currentView.videoId} />,
            ];
          case 'lab':
            return [
              t('routes:lab_title'),
              t('routes:lab_subtitle'),
              t('routes:lab_helper'),
              // eslint-disable-next-line react/jsx-key
              <Lab />,
            ];
          case 'analytics':
            return [
              t('routes:analytics_title'),
              t('routes:analytics_subtitle'),
              t('routes:analytics_helper_text'),
              // eslint-disable-next-line react/jsx-key
              <AnalyticsPage />,
            ];
          case 'gemCollection':
          default:
            return [
              t('routes:gem_collection_title'),
              t('routes:gem_collection_subtitle'),
              t('routes:gem_collection_helper_text'),
              // eslint-disable-next-line react/jsx-key
              <GemCollection />,
            ];
        }
      }
    }
  }, [currentView, profile, accountLinkCompleted, auth]);

  return (
    <Grid
      container
      alignContent="flex-start"
      style={{
        minHeight: '100%',
      }}
    >
      <Grid
        item
        xs={12}
        style={{
          backgroundColor: theme.palette.background.default,
          paddingTop: profile ? 0 : theme.spacing(12),
        }}
      >
        {currentView.view === 'labEdit' ? (
          <Typography
            variant="h3"
            component="h1"
            color="textSecondary"
            className={classes.labEditTitle}
          >
            <Link onClick={doUpdateCurrentView({ view: 'lab' })}>
              <ArrowBackIcon />
            </Link>
            {currentViewLabel}
          </Typography>
        ) : !accountLinkCompleted ? (
          <Typography
            component="h1"
            color="primary"
            className={classes.congratsTitle}
          >
            {currentViewLabel}
          </Typography>
        ) : (
          <Typography
            variant="h3"
            component="h1"
            color="textSecondary"
            style={{
              whiteSpace: 'pre-line',
              paddingTop: theme.spacing(1),
            }}
          >
            {currentViewLabel}
          </Typography>
        )}
        <Box display="flex" alignItems="center">
          <Typography
            display="inline"
            variant="subtitle1"
            color="textPrimary"
            gutterBottom={true}
            style={{
              marginRight: theme.spacing(1),
              marginBottom: theme.spacing(2),
            }}
          >
            {currentViewSubtitle}
          </Typography>
          {helperText ? <HelperPopover text={helperText} /> : null}
        </Box>

        <Divider light />
      </Grid>

      <Grid
        item
        xs={12}
        style={{
          paddingTop: theme.spacing(4),
        }}
      >
        {currentViewContent}
      </Grid>
    </Grid>
  );
};

const withQueries = declareQueries({
  currentView,
  profile: localProfile,
  auth,
  accountLinkCompleted,
});

export const Dashboard = withQueries(({ queries }): React.ReactElement => {
  return pipe(
    queries,
    QR.fold(
      LazyFullSizeLoader,
      ErrorBox,
      ({ currentView, profile, auth, accountLinkCompleted }) => {
        const classes = useStyles();

        return (
          <Grid container className={classes.root} spacing={4}>
            <Grid item sm={12} md={3} lg={2}>
              <Sidebar
                currentView={currentView}
                accountLinkCompleted={accountLinkCompleted}
                profile={profile}
              />
            </Grid>
            <Grid item sm={12} md={9} lg={10}>
              <DashboardContent
                currentView={currentView}
                profile={profile}
                auth={auth}
                accountLinkCompleted={accountLinkCompleted}
              />
            </Grid>
          </Grid>
        );
      }
    )
  );
});
