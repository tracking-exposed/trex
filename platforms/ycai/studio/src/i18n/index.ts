import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import en from './en-US';


export default i18n.use(initReactI18next).init({
  resources: {
    en: en,
  },
  fallbackLng: 'en',
  interpolation: {
    escapeValue: false,
  },
});
