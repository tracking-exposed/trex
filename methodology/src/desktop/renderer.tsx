import * as React from "react";
import * as ReactDOM from "react-dom";
import { ThemeProvider } from "@material-ui/core";
import { App } from "./App";
import { theme } from "./theme";
import debug from "debug";

const main = (): void => {
  debug.enable("@yttrex*");

  ReactDOM.render(
    <React.StrictMode>
      <ThemeProvider theme={theme}>
        <App />
      </ThemeProvider>
    </React.StrictMode>,
    document.getElementById("guardoni")
  );
};

main();
