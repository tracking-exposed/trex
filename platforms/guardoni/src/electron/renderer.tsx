import { CssBaseline, ThemeProvider } from '@material-ui/core';
import debug from 'debug';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { HashRouter } from 'react-router-dom';
import { App } from './app/App';
import { theme } from './app/theme';

const main = (): void => {
  debug.enable('@trex*');

  ReactDOM.render(
    <React.StrictMode>
      <HashRouter>
        <ThemeProvider theme={theme}>
          <CssBaseline />
          <App />
        </ThemeProvider>
      </HashRouter>
    </React.StrictMode>,
    document.getElementById('guardoni')
  );
};

main();
