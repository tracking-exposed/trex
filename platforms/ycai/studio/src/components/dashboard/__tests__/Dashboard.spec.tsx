import React from 'react';
import renderer from 'react-test-renderer';
import { Dashboard } from '../Dashboard';
import '../../../i18n';
import { ThemeProvider } from '@material-ui/core';
import { YCAITheme } from '../../../theme';

describe('Dashboard', () => {
  test('Should mount the dashboard', () => {
    const component = renderer.create(
      <ThemeProvider theme={YCAITheme}>
        <Dashboard />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
