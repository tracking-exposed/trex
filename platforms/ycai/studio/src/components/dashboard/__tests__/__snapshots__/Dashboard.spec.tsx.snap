// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Should mount the dashboard 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100%",
      "justifyContent": "center",
      "minHeight": 600,
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  <div
    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
